tests/sentry/search/events/builder/test_metrics.py

```diff
@@ -1601,6 +1618,99 @@ def test_invalid_event_type_query_condition(self):
+            value=value,
+            metric=self.ON_DEMAND_KEY_MAP[metric_type],
+            internal_metric=self.ON_DEMAND_MRI_MAP[metric_type],
+            entity=self.ON_DEMAND_ENTITY_MAP[metric_type],
+            tags={
+                **spec_tags,
+                **additional_tags,  # Additional tags might be needed to override field values from the spec.
+            },
+            timestamp=timestamp,
+        )
+
+        return spec
+
+    def test_run_metrics_query_builder_with_groupbys(self):
+        selected_columns = ["count()", "epm()"]
+        duration = 100
+        query_s = f"transaction.duration:>={duration}"
+        groupbys = ["customtag1", "customtag2"]
+        for field in selected_columns:
+            spec = OnDemandMetricSpec(field=field, query=query_s, groupbys=groupbys)
+            # print(spec._query_str_for_hash)
+            # assert (
+            #     spec._query_str_for_hash
+            #     == f"on_demand_{field};{'name': 'event.duration', 'op': 'gte', 'value': 100.0};['customtag1']"
+            # )
+
+            for hour in range(0, 5):
+                self.store_on_demand_metric(
+                    value=hour * 10,
+                    spec=spec,
+                    # Spec tags for fields need to be overriden since the stored value is dynamic
+                    additional_tags={"customtag1": "div > text"},
+                    timestamp=self.start + datetime.timedelta(hours=hour),
+                )
+                self.store_on_demand_metric(
+                    value=hour * 10,
+                    spec=spec,
+                    # Spec tags for fields need to be overriden since the stored value is dynamic
+                    additional_tags={"customtag2": "div > text"},
+                    timestamp=self.start + datetime.timedelta(hours=hour),
+                )
+                self.store_transaction_metric(
+                    value=hour * 10,
+                    tags={"customtag1": "div > text"},
+                    timestamp=self.start + datetime.timedelta(hours=hour),
+                )
+
+        query = MetricsQueryBuilder(
+            self.params,
+            query=query_s,
+            dataset=Dataset.PerformanceMetrics,
+            selected_columns=selected_columns + groupbys,
+            config=QueryBuilderConfig(use_metrics_layer=False, on_demand_metrics_enabled=True),
+        )
+        # The custom tags are not be in the map since they're not aggregates
+        assert query._on_demand_metric_spec_map == {
+            "count()": OnDemandMetricSpec(field="count()", query=query_s, groupbys=groupbys),
+            "epm()": OnDemandMetricSpec(field="epm()", query=query_s, groupbys=groupbys),
+        }
+
+        result = query.run_query("test_query")
+
+        expected = {
+            "data": [
+                {
+                    "c:transactions/on_demand@none": float(duration),
+                    "customtag1": "event.tags.customtag1",
```

Comment:
If the overriden tags work above, this should be returning `customtag1: div > text` not `event.tags.customtag1` (since that is the field, which comes from the spec)